lib/UA.js

```diff
@@ -129,7 +120,17 @@ function UA(uaString) {
 	if (normalized) {
 		this.ua = new useragent.Agent(normalized[1], normalized[2], (normalized[3] || 0), (normalized[4] || 0));
 	} else {
-		this.ua = useragent.parse(stripiOSWebViewBrowsers(uaString));
+
+		// Chrome and Opera on iOS uses a UIWebView of the underlying platform to render
+		// content, by stripping the CriOS or OPiOS strings the useragent parser will alias the
+		// user agent to ios_saf for the UIWebView, which is closer to the actual renderer
+		uaString = uaString.replace(/((CriOS|OPiOS)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|(FxiOS\/(\d+)\.(\d+)))/, '');
+
+		// Vivaldi browser is recognised by UA module but is actually identical to Chrome, so
+		// the best way to get accurate targeting is to remove the vivaldi token from the UA
+		uaString = uaString.replace(/ vivaldi\/[\d\.]+\d+/i, '');
```

Comment:
Good point.  I'd forgotten we could use functions for aliasing.  And it looks like we're already using this for Facebook for iOS, but there's an open issue for Facebook for Android: #990. I'm not really convinced that the function approach works, because it requires that we act on the initial parsed results of the UA string - and in the case of facebook for iOS, that's sort of OK because we can use what the ua module reports as the OS version, as if it's ios_saf.  That's a bit weird though.

Ideally, you want to say 'if this UA is identified as facebook, strip the bit that identifies it as facebook and re-parse it', or even better, turn off the bit of the ua module that matches facebook, in the hope that it would then match chrome or safari instead.

Failing that though, the most reliable option seems to be to strip the Facebook token before we parse it.